### DISCUSSION TITLE
Docker environment variables: change machine learning hostname

### DISCUSSION THREAD

**Discussion by github-cli** [immich-app/immich#6336] [Q&A] [answered]
For some connectivity reasons I want to manually define the hostname of the microservices and machine learning container
I did not see a variable to do that, is this possible or is e.g. immich-machine-learning hardcoded?

For the curious: I have three networks connected, the server and the ML/MS containers share two networks and I want to give an alias on the one they should use to communicate with eachother to stop them from communicating via the undesired one

**Comment by bo0tzz** [comment, marked as answer]
The machine-learning address can be changed via the immich settings. Those can be configured either in the ui or via a config file.

**Comment by github-cli** [comment]
and the microservices container?

**Reply by bo0tzz** [reply]
It isn't communicated with over HTTP, so there's nothing to change there.

**Reply by github-cli** [reply]
but they communicate somehow I am guessing?
if it needs to reach it via a specific name, how do I change that?
Or is everything done via database/folder access?

**Reply by bo0tzz** [reply]
They communicate through the database and redis.

**Reply by github-cli** [reply]
awesome, good to know
does the microservices container need internet access even or can I completely isolate the container?
Internet access is only needed for the server (I am assuming to download files?) and the machine learning container (to download models), yes?
